src/Symfony/Component/Clock/DatePoint.php

```diff
@@ -66,6 +66,30 @@ public static function createFromMutable(\DateTime $object): static
         return parent::createFromMutable($object);
     }
 
+    public static function createFromTimestamp(int|float $timestamp): static
+    {
+        if (\PHP_VERSION_ID >= 80400) {
+            return parent::createFromTimestamp($timestamp);
+        }
+
+        if (\is_int($timestamp)) {
+            return (new static())->setTimestamp($timestamp);
```

Comment:
should be a bit faster:
```suggestion
            return new static(reference: new \DateTimeImmutable('@'.$timestamp));
```

Reply:
Is it? I wanted to avoid casting to string and back. Did you run a benchmark?

Reply:
I didn't but I'm pretty confident give new static calls clock::get to only trash the value just after

Reply:
Oh, right. I've benchmarked a couple of solutions now and this one was the fastest on my machine:

```suggestion
            return static::createFromFormat('U', (string) $timestamp);
```

The script I used: 

```php
namespace Symfony;

use Symfony\Component\Clock\DatePoint;
use Symfony\Component\Stopwatch\Stopwatch;

require __DIR__ .'/vendor/autoload.php';

$stopwatch = new Stopwatch();
$stopwatch->start('date-point');

for ($i = 0; $i < 1_000_000; $i++) {
    DatePoint::createFromTimestamp($i);
}

dump($stopwatch->stop('date-point')->getDuration());
```